app/src/main/java/org/cru/godtools/fragment/ToolDetailsFragment.java

```diff
@@ -332,6 +344,30 @@ void removeTool() {
         }
     }
 
+    @Optional
+    @OnClick(R.id.action_open)
+    void openTool() {
+        if (mTool != null && mTool.getCode() != null) {
+            Settings settings = Settings.getInstance(requireContext());
+            Locale primaryLanguage = settings.getPrimaryLanguage();
+            Locale parallelLanguages = settings.getParallelLanguage();
+            switch (mTool.getType()) {
+                case TRACT:
+                    // start preLoading the tract in the first language
+                    ManifestManager.getInstance(requireContext())
+                            .getLatestPublishedManifest(mTool.getCode(), primaryLanguage);
+
+                    TractActivity.start(requireActivity(), mTool.getCode(), primaryLanguage,
+                                        parallelLanguages);
+                    break;
+                case ARTICLE:
+                    CategoriesActivityKt.startCategoriesActivity(requireActivity(), mTool.getCode(),
+                                                                 primaryLanguage);
+                    break;
+            }
```

Comment:
I'm not a huge fan of duplicating this logic here, what are the options available to be able to re-use this between the main tool list and this fragment?

Reply:
OK, I will look to see if I can find a way.